tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -145,6 +144,53 @@ public async Task InstantJobShouldPassDownParameter()
         content.ShouldBe("Hello from InstantJob");
     }
 
+    [Theory]
+    [MemberData(nameof(InstantJobRunners))]
+    public async Task InstantJobCanStartADisabledJob(Func<IInstantJobRegistry, string, CancellationToken, Guid> instantJobRunner)
```

Comment:
Reintegrated from #185.

The various fixes introduced in the first commit allows this test to not hang anymore

Reply:
Incidentally, this "hanging" while triggering instant runs against disabled jobs has been a blocker for me at $DAYJOB to run integration tests.

Reply:
I like your `$DAYJOB` ;-)